server/scratchy_server/app.py

```diff
@@ -30,4 +44,15 @@
 api.add_resource(RoomRes, '/api/room', '/api/room/<string:roomId>')
 api.add_resource(UserRes, '/api/user', '/api/user/<string:userId>')
 api.add_resource(MessageRes, '/api/message', '/api/message/<string:messageId>')
+
+# path for the apispec you can have info there: https://flask-apispec.readthedocs.io/en/latest/usage.html
+app.add_url_rule('/api/room/<string:roomId>', view_func=RoomRes.as_view('RoomRes'))
+app.add_url_rule('/api/user/<string:userId>', view_func=UserRes.as_view('UserRes'))
+app.add_url_rule('/api/message/<string:messageId>', view_func=MessageRes.as_view('MessageRes'))
```

Comment:
I'm not sure it comes from here but `post` request don't need ID : 
![post](https://user-images.githubusercontent.com/1092728/116921003-ab3b8600-ac53-11eb-9393-79f2e68c9929.png)

Can you take a look at it ? maybe a decorator to unset path parameters on the post method is needed ?

Reply:
We are not correctly using flaskrestfull see the full example in their documentation : https://flask-restful.readthedocs.io/en/latest/quickstart.html#full-example

We need to have 2 separate ressources with 2 route, 1 ressource with the ressourceId in the route containing GET/PUT/DELETE and another ressource without id in the route path containing POST/GET(ALL)

![flaskApiRestFull](https://user-images.githubusercontent.com/1092728/117074366-7ef73680-ad33-11eb-8a3a-cb3ff41d90ce.png)

This should solve this issue, @Draeggon can you do this ?

Reply:
Solved.